core/src/main/java/org/bitcoinj/core/Transaction.java

```diff
@@ -87,29 +87,25 @@
  * <p>Instances of this class are not safe for use by multiple threads.</p>
  */
 public class Transaction extends ChildMessage {
+    private static final Comparator<Transaction> SORT_TX_BY_ID = Comparator.comparing(Transaction::getTxId);
     /**
      * A comparator that can be used to sort transactions by their updateTime field. The ordering goes from most recent
      * into the past.
      */
-    public static final Comparator<Transaction> SORT_TX_BY_UPDATE_TIME = (tx1, tx2) -> {
-        Instant t1 = tx1.getUpdateTimeInstant().orElse(Instant.EPOCH);
-        Instant t2 = tx2.getUpdateTimeInstant().orElse(Instant.EPOCH);
-        final int updateTimeComparison = -(t1.compareTo(t2));
-        //If time1==time2, compare by tx hash to make comparator consistent with equals
-        return updateTimeComparison != 0 ? updateTimeComparison : tx1.getTxId().compareTo(tx2.getTxId());
-    };
+    public static final Comparator<Transaction> SORT_TX_BY_UPDATE_TIME =
+            Comparator.<Transaction, Instant>comparing(
+                            tx -> tx.getUpdateTimeInstant().orElse(Instant.EPOCH),
+                            Comparator.reverseOrder())
+                    .thenComparing(SORT_TX_BY_ID);
     /** A comparator that can be used to sort transactions by their chain height. */
-    public static final Comparator<Transaction> SORT_TX_BY_HEIGHT = (tx1, tx2) -> {
-        final TransactionConfidence confidence1 = tx1.getConfidence();
-        final int height1 = confidence1.getConfidenceType() == ConfidenceType.BUILDING
-                ? confidence1.getAppearedAtChainHeight() : Block.BLOCK_HEIGHT_UNKNOWN;
-        final TransactionConfidence confidence2 = tx2.getConfidence();
-        final int height2 = confidence2.getConfidenceType() == ConfidenceType.BUILDING
-                ? confidence2.getAppearedAtChainHeight() : Block.BLOCK_HEIGHT_UNKNOWN;
-        final int heightComparison = -(Integer.compare(height1, height2));
-        //If height1==height2, compare by tx hash to make comparator consistent with equals
-        return heightComparison != 0 ? heightComparison : tx1.getTxId().compareTo(tx2.getTxId());
-    };
+    public static final Comparator<Transaction> SORT_TX_BY_HEIGHT =
+            Comparator.<Transaction, Integer>comparing(
+                            tx -> tx.getConfidence().getConfidenceType() == ConfidenceType.BUILDING ?
+                                    tx.getConfidence().getAppearedAtChainHeight() :
```

Comment:
On these lines I'm get `getConfidence()` twice. Previously, I was caching it in a local variable. How can I do this with functions?

Reply:
See below...

Reply:
Note that you can also use the `return` form Lambda:
```
tx -> {
   var cache = xyz;
   computation;
    return <integer>
}